src/pykeen/evaluation/evaluation_loop.py

```diff
@@ -101,16 +102,21 @@ def __init__(
         model: Model,
         dataset: Dataset[BatchType],
         evaluator: Evaluator,
+        mode: InductiveMode | None = None,
     ) -> None:
         """Initialize the evaluation loop.
 
         :param model: the model to evaluate.
         :param dataset: the evaluation dataset
         :param evaluator: the evaluator instance
+        :param mode: the inductive mode for evaluation. Defaults to None, meaning transductive setting
         """
         self.model = model
         self.evaluator = evaluator
         self.dataset = dataset
+        if mode is not None:
+            raise NotImplementedError(f"non-transductive evaluation mode is not yet implemented: {mode}")
```

Comment:
The evaluation loop supports it. The only issue is that we don't expose the mode through the `pipeline`. However, this may be acceptable since adding support for inductive models to the `pipeline` could make it more difficult to determine which parameter combinations to provide.

Reply:
thanks! I noticed I misunderstood this and updated it after the first commit.

There was definitely some discussion somewhere about how to document exposing the inductive mode via the pipeline (https://github.com/pykeen/pykeen/pull/1503#discussion_r1983978912)

I wouldn't block up this PR with solving that problem